In [29]:
# import the model prediction code
from myModel import *

# initialize the model
myModel = Zhai_predictLymph()

"""
    Calculate the risk score based on the input data using the Cox hazard formula.
 
    Takes JSON patient file as input.
    
    Returns:
        dict: A dictionary containing:
            - risk_score (str): Calculated risk score as a percentage string
            - risk_category (str): Risk classification (High risk, Medium risk, Low risk)
    """


# Sample Patient 1: High Risk
# Male, T4 stage, WHO PS 4, Large node (5.0cm), High heterogeneity (0.9)
high_risk_patient = {
    "resourceType": "Bundle",
    "type": "collection",
    "entry": [
        {
            "resource": {
                "resourceType": "Patient",
                "extension": [
                    {
                        "url": "http://hl7.org/fhir/us/core/StructureDefinition/us-core-birthsex",
                        "valueCodeableConcept": {
                            "coding": [{"code": "248153007", "system": "https://www.snomed.org/"}]
                        }
                    }
                ]
            }
        },
        {
            "resource": {
                "resourceType": "Condition",
                "stage": [{
                    "summary": {
                        "coding": [{"code": "1228888003", "system": "https://www.snomed.org/"}]
                    }
                }]
            }
        },
        {
            "resource": {
                "resourceType": "Observation",
                "code": {"coding": [{"code": "C1298651", "system": "http://loinc.org"}]},
                "valueCodeableConcept": {
                    "coding": [{"code": "373801007", "system": "https://www.snomed.org/"}]
                }
            }
        },
        {
            "resource": {
                "resourceType": "Observation",
                "code": {"coding": [{"code": "least-axis-length-llln"}]},
                "valueQuantity": {"value": 5.0, "unit": "cm"}
            }
        },
        {
            "resource": {
                "resourceType": "Observation",
                "code": {"coding": [{"code": "correlation-glcm"}]},
                "valueDecimal": 0.9
            }
        }
    ]
}

# Sample Patient 2: Medium Risk
# Female, T3 stage, WHO PS 1, Medium node (2.5cm), Moderate heterogeneity (0.5)
medium_risk_patient = {
    "resourceType": "Bundle",
    "type": "collection",
    "entry": [
        {
            "resource": {
                "resourceType": "Patient",
                "extension": [
                    {
                        "url": "http://hl7.org/fhir/us/core/StructureDefinition/us-core-birthsex",
                        "valueCodeableConcept": {
                            "coding": [{"code": "248152002", "system": "https://www.snomed.org/"}]
                        }
                    }
                ]
            }
        },
        {
            "resource": {
                "resourceType": "Condition",
                "stage": [{
                    "summary": {
                        "coding": [{"code": "1228889006", "system": "https://www.snomed.org/"}]
                    }
                }]
            }
        },
        {
            "resource": {
                "resourceType": "Observation",
                "code": {"coding": [{"code": "C1298651", "system": "http://loinc.org"}]},
                "valueCodeableConcept": {
                    "coding": [{"code": "373798008", "system": "https://www.snomed.org/"}]
                }
            }
        },
        {
            "resource": {
                "resourceType": "Observation",
                "code": {"coding": [{"code": "least-axis-length-llln"}]},
                "valueQuantity": {"value": 2.5, "unit": "cm"}
            }
        },
        {
            "resource": {
                "resourceType": "Observation",
                "code": {"coding": [{"code": "correlation-glcm"}]},
                "valueDecimal": 0.5
            }
        }
    ]
}

# Sample Patient 3: Low Risk
# Female, T1 stage, WHO PS 0, Small node (2.0cm), Low heterogeneity (0.2)
low_risk_patient = {
    "resourceType": "Bundle",
    "type": "collection",
    "entry": [
        {
            "resource": {
                "resourceType": "Patient",
                "extension": [
                    {
                        "url": "http://hl7.org/fhir/us/core/StructureDefinition/us-core-birthsex",
                        "valueCodeableConcept": {
                            "coding": [{"code": "248152002", "system": "https://www.snomed.org/"}]
                        }
                    }
                ]
            }
        },
        {
            "resource": {
                "resourceType": "Condition",
                "stage": [{
                    "summary": {
                        "coding": [{"code": "1228889006", "system": "https://www.snomed.org/"}]
                    }
                }]
            }
        },
        {
            "resource": {
                "resourceType": "Observation",
                "code": {"coding": [{"code": "C1298651", "system": "https://loinc.org"}]},
                "valueCodeableConcept": {
                    "coding": [{"code": "373797003", "system": "https://www.snomed.org/"}]
                }
            }
        },
        {
            "resource": {
                "resourceType": "Observation",
                "code": {"coding": [{"code": "least-axis-length-llln"}]},
                "valueQuantity": {"value": 2.0, "unit": "cm"}
            }
        },
        {
            "resource": {
                "resourceType": "Observation",
                "code": {"coding": [{"code": "correlation-glcm"}]},
                "valueDecimal": 0.4
            }
        }
    ]
}

patients = [
    ("High Risk", high_risk_patient),
    ("Medium Risk", medium_risk_patient),
    ("Low Risk", low_risk_patient)
]

for name, bundle in patients:
    result = myModel.predict(bundle)
    print(f"Result patient {name}:")
    print(f"  Risk score: {result.get('risk_score')}")
    print(f"  Category:   {result.get('risk_category')}")



Result patient High Risk:
  Risk score: 100.0%
  Category:   High risk
Result patient Medium Risk:
  Risk score: 13.75%
  Category:   Medium risk
Result patient Low Risk:
  Risk score: 2.52%
  Category:   Low risk


Output for this high risk test patient is: 
Risk score: 100%
this means 100% chance of a node will persist or recur after treatment.
according to the paper >60% is high risk.

Output for this medium risk test patient is: 
Risk score: 13.75%
this means 13.75% chance of a node will persist or recur after treatment.
according to the paper between 10% and 60% is medium risk.

Output for this low risk test patient is: 
Risk score: 2.52%
this means 2.52% chance of a node will persist or recur after treatment.
according to the paper <10% is low risk.

